doc/user-guide.adoc

```diff
@@ -29,6 +29,147 @@ The `applicationImage` accepts different references to Docker image:
+`helloworld-rs`. We provide the `MAVEN_ARGS`and `ARTIFACT_DIR` environment variables to S2I to build only this Web application.
+
+Once the application image is build and deployed, the Operator will install it on the OpenShift cluster and create a Route for it.
+The application can then be accessed with:
+
+[source,shell]
+----
+$ curl $(oc get wfly/quickstart -o jsonpath='{.status.hosts[0]}')/helloworld-rs/rest/json
+{"result":"Hello World!"}
+----
+
+[[source-2-image]]
+### Specify Source2Image Configuration
+
+To build the application image, the WildFly S2I images must be installed in the OpenShift cluster before the WildFly Operator deploys the application.
+As an example, to install the WildFly 19 S2I images in the `openshift` cluster, you can run the commands:
+
+[source,shell]
+----
+$ oc import-image wildfly/wildfly-centos7:19.0 --from=quay.io/wildfly/wildfly-centos7:19.0 -n openshift --confirm
+$ oc import-image wildfly/wildfly-runtime-centos7:19.0 --from=quay.io/wildfly/wildfly-runtime-centos7:19.0 -n openshift --confirm
+----
+
+To be able to build an application from its source, the `builderImage`
+must be set in the `source2Image` section. It must correspond to an Image Stream tag of the WildFly S2I Builder image stream.
+
+Optionally, if the `runtimeImage` is set, a 2nd application image will be built that only contains the runtime parts required to run the application on WildFly. This `runtimeImage` must correspond to an Image Stream tag of the WildFly S2I Runtime image stream.
+
+If an image stream tag referenced by a `WildFlyServer` resource is updated (for example, when a new image fixes a CVE), the Operator will detect this image change and trigger the build of a new applicaiton image and deploy it on OpenShift.
+
+[[source-webhook]]
+### Specify WebHook for Application Code changes
+
+When the Operator deploys an application that is configured with `applicationSource`, it provide Webhooks to trigger new builds of the application image upon code changes.
+The `WildFlyServer` custom resourc definition supports GitHub and Generic Webhooks. For more information on these Webhooks, please consult the latest https://docs.openshift.com/container-platform/4.3/builds/triggering-builds-build-hooks.html[OpenShift documentation].
+
+The GitHub and Generic Webhooks relies on secret with a key named `WebHookSecretKey` whose value is supplied when invoking the webhook.
+When a `WildFlyServer` is created, you can specify the secret that holds this key with either `gitHubWebHookSecret` or `genericHubWebHookSecret`. If these fields are not specified, the Operator will automatically create two secrets named `<wildflyserver name>-github-webhook` and `<wildflyserver name>-generic-webhook` in the same namespace than the `WildFlyServer` resource and use a random string as the value of their `WebHookSecretKey`.
```

Comment:
`in the same namespace than the WildFlyServer resource` -> `in the same namespace as the WildFlyServer resource` ?